### ISSUE TITLE
Wiring

### ISSUE THREAD

**Issue by DanielMarcato** [devmapal/esp8266-thermostat-control#1]
Can you please show the wiring of the encoder pins? In the code there are in and output pins that somehow are both marked as pins for the rotary encoder. I thought the encoder only has pin1, pin2 and GND

**Comment by GamerClassN7** [comment]
> Can you please show the wiring of the encoder pins? In the code there are in and output pins that somehow are both marked as pins for the rotary encoder. I thought the encoder only has pin1, pin2 and GND

I think he is using pins both like input and output in same time to be able to send puls to set temperature and read puls to get temperature when set manually.

**Comment by devmapal** [comment]
> > Can you please show the wiring of the encoder pins? In the code there are in and output pins that somehow are both marked as pins for the rotary encoder. I thought the encoder only has pin1, pin2 and GND
> 
> I think he is using pins both like input and output in same time to be able to send puls to set temperature and read puls to get temperature when set manually.

@GamerClassN7 Exactly. I've wired the ESP8266 between the pcb rotary encoder input pins of the thermostat and the actual rotary encoder. one set of pins reads from the dial, the other set of pins sends input signals to the pcb. @DanielMarcato Let me know if that answers your question, otherwise I'm happy to add more details

**Comment by GamerClassN7** [comment]
@devmapal  Hi, thanks for answer :) I am working on littele bit larger in scope Smart Home Automation Project :)  link: https://github.com/Simple-Home. If you would want participate (by any miracle let me know :)) :). But back to your project are you running ESP from internal battery ? i am bud i cant get longer than 9h because i am using ESP 01 there is no possibility of deep sleep did you run into same problem ? because if i am able to run it form internal battery i can gide hole system in original case :) Thanks for any answer in advance :)

**Comment by devmapal** [comment]
@GamerClassN7 In fact I've into the same issue. There is a deeper slee

**Comment by devmapal** [comment]
@GamerClassN7 I'vs run into exactly the same issue. There is a deeper sleep mode the ESP8266 supports which would allow it to run for years. In this mode it doesn't process interrupts anymore, so you would loose the ability to manually set the temperature.

Another alternative would be to use a second microcontroller to control the rotary encoder that can go into a deep enough sleep state but still be woken up by interrupts. Then you could hook up both the manual controls as well as the ESP to it. There are many microcontrollers e.g. from Atmel that support this. It does make the design quite a bit more complex though.

A third option (which I went for for now) is simply hook the thermostat up to a powersocket :-)

**Comment by GamerClassN7** [comment]
@devmapal HI, i find a weary word problem for no reason my thermostat goes to 17 degree, how i switch termostat to fully manual mode ? I am suspicious of some smart feature :D

**Comment by GamerClassN7** [comment]
@devmapal  i run into the problem with ESP 01 i wont boot when conected to termostatic valve did you run into same issue ?  what type of board did you used
